### PR TITLE
Run flake8 tests to find Python syntax errors and undefined names

### PR DESCRIPTION
**Pull request by cclauss**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -13,6 +13,10 @@ jobs:
   - script: |
       source install.sh
     displayName: 'Install nni toolkit via source code'
+  - script: |
+      python3 -m pip install flake8
+      flake8 . --count --select=E9,F63,F72,F82 --show-source --statistics
```
Comment (xuehui1991) [inline azure-pipelines.yml]: you may need to add "--user"

**Comment by chicm-ms** [comment]
@cclauss  Thanks for contributing this PR. 
There are some annotation related errors can not be fixed right now, those annotated py files are executed after the annotation is processed.  Could you please add following files as ignored files?
```
./tools/nni_annotation/*:F821
./examples/trials/mnist-nas/mnist.py:F821
```

The rest errors can be fixed soon, we will merge this PR after the errors are fixed.

**Comment by chicm-ms** [comment]
@cclauss  The wildcard `./tools/nni_annotation/*:F821` seems not working,  following are the files I would like to ignore:
```
./tools/nni_annotation/testcase/mutable_layer_usercode/simple.py:F821
./tools/nni_annotation/testcase/usercode/dir/simple.py:F821
./tools/nni_annotation/testcase/annotated/dir/simple.py:F821
./tools/nni_annotation/testcase/annotated/handwrite.py:F821
./examples/trials/mnist-nas/mnist.py:F821
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -13,6 +13,11 @@ jobs:
   - script: |
       source install.sh
     displayName: 'Install nni toolkit via source code'
+  - script: |
+      python3 -m pip install flake8 --user
+      IGNORE=./tools/nni_annotation/testcase:F821,./examples/trials/mnist-nas/mnist.py:F821
```
Comment (chicm-ms) [inline azure-pipelines.yml]: probably should use --per-file-ignores ?

**Comment by chicm-ms** [comment]
@cclauss Thanks,  the ignore config is working now.

**Pull request merged**